popup.js

```diff
@@ -0,0 +1,251 @@
+        function doneLoading() {
+            var hide = document.getElementsByClassName('loading');
+            hide[0].style.display = 'none';
+            var loadStatus = document.getElementsByClassName('status');
+            loadStatus[0].style.display = 'none';
+            var prgbr = document.getElementById('container');
+            prgbr.display = 'block';
+        }
+
+
+
+
+
+
+
+        function initBar() {
+            var progress = document.getElementById('container');
+            progress.style.display = 'block';
+        }
+
+        function initSuggestion() {
+            var butt = document.getElementById('suggestions');
+            butt.style.display = 'block';
+        }
+
+        function initBreakdown() {
+            var butt2 = document.getElementById('breakdown');
+            butt2.style.display = 'block';
+        }
+
+
+
+
+
+
+
+
+
+        //progress bar stuff
+        var bar = new ProgressBar.SemiCircle(container, {
+            strokeWidth: 6,
+            color: '#FFEA82',
+            trailColor: '#eee',
+            trailWidth: 1,
+            easing: 'easeInOut',
+            duration: 1400,
+            svgStyle: null,
+            text: {
+                value: '',
+                alignToBottom: false
+            },
+            from: { color: '#FF533D' },
+            to: { color: '#4FC1A3' },
+            // Set default step function for all animate calls
+            step: (state, bar) => {
+                bar.path.setAttribute('stroke', state.color);
+                var value = Math.round(bar.value() * 100);
+                if (value === 0) {
+                    bar.setText('');
+                } else {
+                    bar.setText(value);
+                }
+
+                bar.text.style.color = state.color;
+            }
+        });
+        bar.text.style.fontFamily = '"Raleway", Helvetica, sans-serif';
+        bar.text.style.fontSize = '2rem';
+
+
+
+
+
+
+         function displayTopPageSpeedSuggestions(result) {
+            var results = [];
+            var ruleResults = result.formattedResults.ruleResults;
+            for (var i in ruleResults) {
+                var ruleResult = ruleResults[i];
+                // Don't display lower-impact suggestions.
+                if (ruleResult.ruleImpact < 3.0) continue;
+                results.push({
+                    name: ruleResult.localizedRuleName,
+                    impact: ruleResult.ruleImpact
+                });
+            }
+            results.sort(sortByImpact);
+
+            var suggestions = document.createElement('h2');
+            var ul = document.createElement('ul');
+            for (var i = 0, len = results.length; i < len; i++) {
+                var r = document.createElement('li');
+                r.innerHTML = results[i].name;
+                ul.insertBefore(r, null);
+            }
+            if (ul.hasChildNodes()) {
+                var be4 = document.getElementById('breakdown');
```

Comment:
sorry I used be4 because it sounds like "before" but it's shorter. I'll change that